src/main.rs

```diff
@@ -676,3 +693,54 @@ fn run_on_boot_linux(interval: &u32) -> Result<(), Box<dyn std::error::Error>> {
 
     Ok(())
 }
+
+#[cfg(target_os = "windows")]
+fn setup_registry() {
+    if registry::registry(
+        get_config_path(CONFIG_DIR, LOGO).to_str().unwrap(),
+        APP_NAME,
+    )
+    .is_ok()
+    {
+        info!("Registry succesfully setup.");
+    } else {
+        error!("Registry setup was unsuccesful");
+    }
+}
+
+async fn download_logo() {
+    let logo_path = get_config_path(CONFIG_DIR, LOGO);
+
+    if logo_path.exists() {
+        return;
+    }
+
+    let logo_url = "https://raw.githubusercontent.com/dhruvan2006/tuenroll/main/logo.png";
+
+    if let Some(parent) = logo_path.parent() {
+        std::fs::create_dir_all(parent).expect("Failed to create log directory");
+    }
+    let _ = std::fs::OpenOptions::new()
+        .write(true)
+        .create(true)
+        .truncate(true)
+        .open(&logo_path)
+        .unwrap();
+
+    let response = reqwest::get(logo_url)
+        .await
+        .expect("Request did not succeed. Try again later.");
+
+    info!("Downloading logo.");
+
+    std::fs::write(
+        &logo_path,
+        response
+            .bytes()
+            .await
+            .expect("Error occured while downloading image bytes"),
+    )
+    .expect("Could not write to file.");
+
+    info!("Writing logo to file");
+}
```

Comment:
Invert also the logo_url and logo_path

Reply:
What do you mean by this?

Reply:
Basically pass it in as fn parameters so you can provide your own mocks